File: java/v1/src/main/java/com/api/v1/mappers/customer/CustomerResponseMapper.java
package com.api.v1.mappers.customer;

import com.api.v1.domain.customer.Customer;
import com.api.v1.dtos.customer.CustomerResponseDto;

public record CustomerResponseMapper(Customer customer) {

    public CustomerResponseDto mapToDto() {
        return new CustomerResponseDto(
                customer.getFullName(),
                customer.getSsn(),
                customer.getBirthDate(),
                customer.getEmail(),
                customer.getAddress(),
                customer.getPhoneNumber(),
                customer.getCreatedAt(),
                customer.getUpdatedAt()
        );
    }

}

File: v2/src/main/kotlin/com/api/v2/customer/mappers/CustomerResponseMapper.kt
package com.api.v2.customer.mappers

import com.api.v2.customer.domain.Customer
import com.api.v2.customer.dtos.CustomerResponseDto

class CustomerResponseMapper {

    companion object {

        fun mapToDto(customer: Customer): CustomerResponseDto {
            return CustomerResponseDto(
                customer.fullName(),
                customer.ssn,
                customer.birthDate,
                customer.email,
                customer.phoneNumber,
                customer.gender,
                customer.createdAt,
                customer.updatedAt
            )
        }
    }

}